types/hapi-auth-cookie/index.d.ts

```diff
@@ -0,0 +1,157 @@
+         * If `true`, automatically sets the session cookie after validation to extend the current session for a new `ttl` duration.
+         * @default `false`.
+         */
+        keepAlive?: boolean;
+        /**
+         * If `false` omitted. Other options `Strict` or `Lax`.
+         * Defaults to Strict
+         */
+        isSameSite?: boolean | isSameSite;
+        /**
+         * If `false`, the cookie is allowed to be transmitted over insecure connections which exposes it to attacks.
+         * @default `true`
+         */
+        isSecure?: boolean;
+        /**
+         * If `false`, the cookie will not include the `HttpOnly` flag.
+         * @default `true`
+         */
+        isHttpOnly?: boolean;
+        /**
+         * Optional login URI or function function(request) that returns a URI to redirect unauthenticated requests to.
+         * Note that it will only trigger when the authentication mode is `required`.
+         * To enable or disable redirections for a specific route, set the route plugins config
+         *      `{
+         *         options: {
+         *           plugins: {
+         *             'hapi-auth-cookie': {
+         *                redirectTo: false
+         *              }
+         *           }
+         *         }
+         *       }`.
+         * @default `undefined` (i.e. no redirection).
+         */
+        redirectTo?: string | ((request: Request) => URL);
+        /**
+         * If `true` and `redirectTo` is `true`, appends the current request path to the query component of the `redirectTo` URI using the parameter name `next`.
+         * Set to a string to use a different parameter name.
+         * @default `false`.
+         */
+        appendNext?: string | boolean;
+        /**
+         * An optional session validation function used to validate the content of the session cookie on each request.
+         * Used to verify that the internal session state is still valid (e.g. user account still exists).
+         * The function has the signature function(request, session).
+         * @param request the `Request` object
+         * @param session is the session object set via `request.cookieAuth.set()`
+         */
+        validateFunc: ValidateFunc;
+        /**
+         * **USE WITH CAUTION** an optional name to use with decorating the request object.
+         * @default `cookieAuth`. Using multiple decorator names for separate authentication strategies could allow a developer to call the methods for the wrong strategy.
+         * Potentially resulting in unintended authorized access.
+         */
+        requestDecoratorName?: string;
+    }
+
+    interface ValidateResponse {
+        valid: boolean;
+        credentials?: AuthCredentials;
+    }
+
+    type ValidateFunc = (request: Request, session: any) => Promise<ValidateResponse>;
+}
+
+declare enum isSameSite {
+    Lax,
+    Strict
+}
+
+declare const hapiAuthCookie: Plugin<hapiAuthCookie.ValidateFunc>;
```

Comment:
Since the primary export of this package isn't callable or constructable (i.e. it's an object with with a register method, not itself a function or a class), I think it would be better to follow [this template](http://www.typescriptlang.org/docs/handbook/declaration-files/templates/module-d-ts.html) rather than the namespace-merging setup you have currently.